File: myoro_matchup/lib/module/game/domain/dto/game_team_response_dto.dart
import 'package:faker/faker.dart';
import 'package:flutter/foundation.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:myoro_flutter_annotations/myoro_flutter_annotations.dart';
import 'game_player_response_dto.dart';

part 'game_team_response_dto.g.dart';

/// Game team response DTO.
@immutable
@myoroModel
@JsonSerializable()
final class GameTeamResponseDto with _$GameTeamResponseDtoMixin {
  /// Default constructor.
  const GameTeamResponseDto({required this.index, required this.players, required this.substitutes});

  /// Fake constructor.
  factory GameTeamResponseDto.fake() {
    final playerCount = faker.randomGenerator.integer(11, min: 1);
    final substituteCount = faker.randomGenerator.integer(5);
    return GameTeamResponseDto(
      index: faker.randomGenerator.integer(10),
      players: List.generate(playerCount, (_) => GamePlayerResponseDto.fake()),
      substitutes: List.generate(substituteCount, (_) => GamePlayerResponseDto.fake()),
    );
  }

  /// From JSON.
  factory GameTeamResponseDto.fromJson(Map<String, dynamic> json) {
    return _$GameTeamResponseDtoFromJson(json);
  }

  /// Team index (0-based, used to generate "Team #1", "Team #2", etc.).
  final int index;

  /// Players in the team.
  final List<GamePlayerResponseDto> players;

  /// Substitutes for the team.
  final List<GamePlayerResponseDto> substitutes;

  /// To JSON.
  Map<String, dynamic> toJson() {
    return _$GameTeamResponseDtoToJson(this);
  }
}

File: myoro_matchup_api_old/src/main/java/com/myoro/myoro_matchup_api/dto/GameTeamResponseDto.java
package com.myoro.myoro_matchup_api.dto;

import java.util.List;

/** Game team response DTO. */
public class GameTeamResponseDto {
  /** Team index (0-based, used to generate "Team #1", "Team #2", etc.). */
  private Integer index;

  /** Players in the team. */
  private List<GamePlayerResponseDto> players;

  /** Substitutes for the team. */
  private List<GamePlayerResponseDto> substitutes;

  /** Default constructor. */
  public GameTeamResponseDto() {}

  /** Constructor with all fields. */
  public GameTeamResponseDto(
      Integer index, List<GamePlayerResponseDto> players, List<GamePlayerResponseDto> substitutes) {
    this.index = index;
    this.players = players;
    this.substitutes = substitutes;
  }

  /**
   * Getter for index
   *
   * @return the team index
   */
  public Integer getIndex() {
    return index;
  }

  /**
   * Getter for players
   *
   * @return the players in the team
   */
  public List<GamePlayerResponseDto> getPlayers() {
    return players;
  }

  /**
   * Getter for substitutes
   *
   * @return the substitutes for the team
   */
  public List<GamePlayerResponseDto> getSubstitutes() {
    return substitutes;
  }

  /** Setter for index. */
  public void setIndex(Integer index) {
    this.index = index;
  }

  /** Setter for players. */
  public void setPlayers(List<GamePlayerResponseDto> players) {
    this.players = players;
  }

  /** Setter for substitutes. */
  public void setSubstitutes(List<GamePlayerResponseDto> substitutes) {
    this.substitutes = substitutes;
  }
}
